app/src/org/commcare/dalvik/activities/FormRecordListActivity.java

```diff
@@ -210,13 +239,38 @@ protected void onStop() {
         saveLastQueryString(this.TAG + "-" + KEY_LAST_QUERY_STRING);
     }
 
+    public void onBarcodeFetch(String result, Intent intent) {
+        setSearchText(result);
+    }
+
+
+    public void onCalloutResult(String result, Intent intent) {
+        if (BuildConfig.DEBUG) {
+            throw new IllegalArgumentException("Callout not implemented!");
+        }
+    }
```

Comment:
Try to avoid implementing an interface if you aren't going to actually implement one of its methods. It is often a sign that the interface abstractions being used aren't necessary or need more refining.